### PR TITLE
feat: tagging jumpstart models

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/constants.py [hunk 1]
```diff
@@ -149,3 +151,10 @@ class VariableScope(str, Enum):
 
     CONTAINER = "container"
     ALGORITHM = "algorithm"
+
+
+class JumpStartTag(str, Enum):
+    """Enum class for tags to apply to JumpStart models."""
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/constants.py:157]: Is that the tag key? If yes, can you update the docstring to make it clear please?

File: src/sagemaker/jumpstart/utils.py [hunk 2]
```diff
@@ -136,3 +138,53 @@ def is_jumpstart_model_input(model_id: Optional[str], version: Optional[str]) ->
...
+    return bucket in constants.JUMPSTART_BUCKET_NAME_SET
+
+
+def add_jumpstart_tags(
+    tags: Optional[List[Dict[str, str]]],
+    inference_model_uri: Optional[str],
+    inference_script_uri: Optional[str],
+) -> List[Dict[str, str]]:
+    """Adds tags for JumpStart models. Returns original tags for non-JumpStart
+    models.
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/utils.py:163]: nit:
```py
"""Add custom tags to JumpStart models, return the updated tags.

No-op if this is not a JumpStart model related resource.

...
"""
```

File: src/sagemaker/jumpstart/utils.py [hunk 3]
```diff
@@ -136,3 +138,53 @@ def is_jumpstart_model_input(model_id: Optional[str], version: Optional[str]) ->
...
+        )
+
+    if is_jumpstart_model_uri(inference_script_uri):
+        if tags is None:
+            tags = []
+        tags.append(
+            {
+                constants.JumpStartTag.INFERENCE_SCRIPT_URI.value: inference_script_uri,
+            }
+        )
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/utils.py:188]: I know this is an edge case, but could you check that these tags do not already exist, and not add them if that's the case? You would get a validation error if you tried to add two tags with the same key.

File: tests/unit/sagemaker/jumpstart/test_utils.py [hunk 4]
```diff
@@ -112,3 +121,168 @@ def test_get_sagemaker_version(patched_parse_sm_version: Mock):
...
+        tags=tags,
+        inference_model_uri=inference_model_uri,
+        inference_script_uri=inference_script_uri,
+    ) == [
+        {"some": "tag"},
+        {
+            JumpStartTag.INFERENCE_MODEL_URI.value: inference_model_uri,
+        },
+        {JumpStartTag.INFERENCE_SCRIPT_URI.value: inference_script_uri},
+    ]
```
Comment (JGuinegagne) [inline tests/unit/sagemaker/jumpstart/test_utils.py:288]: could you please add a unit test for the edge case where these tag keys are requested by the customer?

File: src/sagemaker/jumpstart/constants.py [hunk 5]
```diff
@@ -149,3 +151,10 @@ class VariableScope(str, Enum):
...
+
+    INFERENCE_MODEL_URI = "jumpstart-inference-model-uri"
+    INFERENCE_SCRIPT_URI = "jumpstart-inference-script-uri"
```
Comment (JGuinegagne) [inline src/sagemaker/jumpstart/constants.py:160]: could you use something like: `aws-jumpstart-` or `aws-smjs-` prefix to make it clear this was added by AWS?

System tags look like `aws:` so I would like to be close to this, even though these are not system tags (see [tagging](https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html)).

**Review by JGuinegagne — CHANGES_REQUESTED** [review]
Few comments:
- tag names
- edge case handling

File: src/sagemaker/jumpstart/utils.py (hunk 3 above)
Comment (evakravi) [reply src/sagemaker/jumpstart/utils.py:188]: Good idea
